src/pages/dataTableUtilities/reducer/tableReducers.js

```diff
@@ -144,6 +144,30 @@ export const toggleShowFinalised = state => {
   return { ...state, data: sortedData, showFinalised: newShowFinalisedState, searchTerm: '' };
 };
 
+export const showIndicators = state => ({ ...state, showIndicators: true });
+
+export const hideIndicators = state => ({ ...state, showIndicators: false });
+
+export const selectIndicator = (state, action) => {
+  const { payload } = action;
+  const { indicatorCode } = payload;
+
+  const { indicators } = state;
+
+  const [selectedIndicator] = indicators.filter(({ code }) => code === indicatorCode);
+
+  const indicatorColumns = selectedIndicator.columns.map(column => ({
+    type: column.valueType,
+    key: column.description,
+    title: column.description,
+    width: 10,
+    sortable: false,
+    editable: false,
+  }));
+
+  return { ...state, selectedIndicator, indicatorColumns };
+};
```

Comment:
second time, depending on if you believe in the principal of two or three, could extract to a utility method

Reply:
3fb6f83a5db933e2ed33e30c267cdb4e7afd8ee2.